src/crawlee/storage_clients/_file_system/_request_queue_client.py

```diff
@@ -0,0 +1,784 @@
+from __future__ import annotations
+
+import asyncio
+import json
+import shutil
+from datetime import datetime, timezone
+from logging import getLogger
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from pydantic import ValidationError
+from typing_extensions import override
+
+from crawlee import Request
+from crawlee._consts import METADATA_FILENAME
+from crawlee._utils.crypto import crypto_random_object_id
+from crawlee._utils.file import atomic_write, json_dumps
+from crawlee.storage_clients._base import RequestQueueClient
+from crawlee.storage_clients.models import AddRequestsResponse, ProcessedRequest, RequestQueueMetadata
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+    from crawlee.configuration import Configuration
+
+logger = getLogger(__name__)
+
+
+class FileSystemRequestQueueClient(RequestQueueClient):
+    """A file system implementation of the request queue client.
+
+    This client persists requests to the file system as individual JSON files, making it suitable for scenarios
+    where data needs to survive process restarts. Each request is stored as a separate file in a directory
+    structure following the pattern:
+
+    ```
+    {STORAGE_DIR}/request_queues/{QUEUE_ID}/{REQUEST_ID}.json
+    ```
+
+    The implementation uses file timestamps for FIFO ordering of regular requests and maintains in-memory sets
```

Comment:
So is this the same mechanism that was used in memory storage request queue previously, or is it new?